packages/react-router/lib/dom/lib.tsx

```diff
@@ -2632,6 +2632,44 @@ export type FetcherWithComponents<TData> = Fetcher<TData> & {
     FetcherFormProps & React.RefAttributes<HTMLFormElement>
   >;
 
+  /**
+   * Loads data from a route. Useful for loading data imperatively inside user
+   * events outside a normal button or form, like a combobox or search input.
+   *
+   * ```tsx
+   * let fetcher = useFetcher()
+   *
+   * <input onChange={e => {
+   *   fetcher.load(`/search?q=${e.target.value}`)
+   * }} />
+   * ```
+   */
+  load: (
+    href: string,
+    opts?: {
+      /**
+       * Wraps the initial state update for this `fetcher.load` in a
+       * [`ReactDOM.flushSync`](https://react.dev/reference/react-dom/flushSync)
+       * call instead of the default [`React.startTransition`](https://react.dev/reference/react/startTransition).
+       * This allows you to perform synchronous DOM actions immediately after the
+       * update is flushed to the DOM.
+       */
+      flushSync?: boolean;
+    },
+  ) => Promise<void>;
+
+  /**
+   * Reset a fetcher back to an empty/idle state.
+   *
+   * If the fetcher is currently in-flight, the
+   * [`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController)
+   * will be aborted with the `reason`, if provided.
+   *
+   * @param reason Optional `reason` to provide to [`AbortController.abort()`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort)
+   * @returns void
+   */
+  unstable_reset: (reason?: unknown) => void;
```

Comment:
I know this is unstable, but what do you think about future proofing this API with an options object since the reason is optional?